File: src/main/java/com/fatec/destino/util/model/usuario/Cpf/CpfConverter.java
package com.fatec.destino.util.model.usuario.Cpf;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter(autoApply = true)
public class CpfConverter implements AttributeConverter<Cpf, String> {

    @Override
    public String convertToDatabaseColumn(Cpf attribute) {
        return attribute == null ? null : attribute.getValorPuro();
    }

    @Override
    public Cpf convertToEntityAttribute(String dbData) {
        return dbData == null ? null : new Cpf(dbData);
    }
}

File: src/main/kotlin/com/fatec/destino/util/model/usuario/cpf/CpfConverter.kt
package com.fatec.destino.util.model.usuario.cpf

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter(autoApply = true)
class CpfConverter : AttributeConverter<Cpf?, String?> {
    override fun convertToDatabaseColumn(attribute: Cpf?): String? {
        return attribute?.getvalorPuro()
    }

    override fun convertToEntityAttribute(dbData: String?): Cpf? {
        return if (dbData == null) null else Cpf(dbData)
    }
}